src/burnchains/tests/db.rs

```diff
@@ -0,0 +1,591 @@
+        500,
+        &canon_hash,
+        &first_bhh,
+        &vec![],
+        485,
+    ));
+    let ops = burnchain_db
+        .store_new_burnchain_block(&burnchain, &headers, &canonical_block)
+        .unwrap();
+    assert_eq!(ops.len(), 0);
+
+    let vtxindex = 1;
+    let noncanon_block_height = 400;
+    let non_canon_hash = BurnchainHeaderHash([2; 32]);
+
+    let fixtures = operations::leader_key_register::tests::get_test_fixtures(
+        vtxindex,
+        noncanon_block_height,
+        non_canon_hash,
+    );
+
+    let parser = BitcoinBlockParser::new(BitcoinNetworkType::Testnet, BLOCKSTACK_MAGIC_MAINNET);
+    let mut broadcast_ops = vec![];
+    let mut expected_ops = vec![];
+
+    for (ix, tx_fixture) in fixtures.iter().enumerate() {
+        let tx = make_tx(&tx_fixture.txstr);
+        let burnchain_tx = parser.parse_tx(&tx, ix + 1).unwrap();
+        if let Some(res) = &tx_fixture.result {
+            let mut res = res.clone();
+            res.vtxindex = (ix + 1).try_into().unwrap();
+            expected_ops.push(res.clone());
+        }
+        broadcast_ops.push(burnchain_tx);
+    }
+
+    let non_canonical_block = BurnchainBlock::Bitcoin(BitcoinBlock::new(
+        400,
+        &non_canon_hash,
+        &first_bhh,
+        &broadcast_ops,
+        350,
+    ));
+
+    let ops = burnchain_db
+        .store_new_burnchain_block(&burnchain, &headers, &non_canonical_block)
+        .unwrap();
+    assert_eq!(ops.len(), expected_ops.len());
+    for op in ops.iter() {
+        let expected_op = expected_ops
+            .iter()
+            .find(|candidate| candidate.txid == op.txid())
+            .expect("FAILED to find parsed op in expected ops");
+        if let BlockstackOperationType::LeaderKeyRegister(op) = op {
+            assert_eq!(op, expected_op);
+        } else {
+            panic!("EXPECTED to parse a LeaderKeyRegister");
+        }
+    }
+
+    let BurnchainBlockData { header, ops } =
+        BurnchainDB::get_burnchain_block(&burnchain_db.conn(), &non_canon_hash).unwrap();
+    assert_eq!(ops.len(), expected_ops.len());
+    for op in ops.iter() {
+        let expected_op = expected_ops
+            .iter()
+            .find(|candidate| candidate.txid == op.txid())
+            .expect("FAILED to find parsed op in expected ops");
+        if let BlockstackOperationType::LeaderKeyRegister(op) = op {
+            assert_eq!(op, expected_op);
+        } else {
+            panic!("EXPECTED to parse a LeaderKeyRegister");
+        }
+    }
+    assert_eq!(&header, &non_canonical_block.header());
+
+    let looked_up_canon = burnchain_db.get_canonical_chain_tip().unwrap();
+    assert_eq!(&looked_up_canon, &canonical_block.header());
+
+    let BurnchainBlockData { header, ops } =
+        BurnchainDB::get_burnchain_block(&burnchain_db.conn(), &canon_hash).unwrap();
+    assert_eq!(ops.len(), 0);
+    assert_eq!(&header, &looked_up_canon);
+}
+
+#[test]
+fn test_classify_stack_stx() {
```

Comment:
Please explain this test. Also, I'm curious how it relates to the Affirmation Maps?

Reply:
It doesn't.  This is an artifact from moving all the tests for this module into their own test module, as part of @kantai's earlier request.